src/main/java/org/folio/consortia/service/impl/HttpRequestServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package org.folio.consortia.service.impl;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import org.folio.consortia.service.HttpRequestService;
+import org.folio.spring.FolioExecutionContext;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class HttpRequestServiceImpl implements HttpRequestService {
+  private final RestTemplate restTemplate;
+  private final FolioExecutionContext folioExecutionContext;
+
+  @Override
+  public ResponseEntity<String> performRequest(String url, HttpMethod httpMethod, Object payload) {
+    FolioExecutionContext currentTenantContext = (FolioExecutionContext) folioExecutionContext.getInstance();
```

Comment:
getOkapiUrl() can be called without calling getInstance()

Reply:
fixed